src/components/Common/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,114 @@
+/**
+ * Navbar.tsx
+ * 
+ * Represents the navigation bar at the top of a webpage.
+ * Includes buttons for menu navigation, a logo, search functionality, user account, and shopping bag.
+ * Handles responsive behavior for different screen sizes.
+ */
+import React, { useState, useEffect } from 'react';
+import { HamburMenu, LogoWithoutName, SearchIcon, UserIcon, BagIcon, ExitIcon } from "@/assets/homePageImg";
+import './Navbar.scss';
+import { MenuPopup, SearchPopup, BagPopup } from "@/components";
+
+const Navbar: React.FC = () => {
+  const [isMenuPopupOpen, setIsMenuPopupOpen] = useState(false);
+  const [isSearchPopupOpen, setIsSearchPopupOpen] = useState(false);
+  const [isBagPopupOpen, setIsBagPopupOpen] = useState(false);
+  const [isDesktop, setIsDesktop] = useState(false);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsDesktop(window.innerWidth > 1024); // Check if screen size is desktop
+    };
+
+    window.addEventListener('resize', handleResize);
+    handleResize();
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+
+  const toggleMenuPopup = () => {
```

Comment:
The functionality of toggling popups could be made more generic to avoid repetition and improve maintainability. You can create a single function to handle toggling of popups and pass parameters to specify which popup to toggle.